plugin/kubernetes/controller.go

```diff
@@ -491,63 +508,176 @@ func (dns *dnsControl) updateModifed() {
 	atomic.StoreInt64(&dns.modified, unix)
 }
 
-func (dns *dnsControl) Add(obj interface{})    { dns.updateModifed() }
-func (dns *dnsControl) Delete(obj interface{}) { dns.updateModifed() }
+func (dns *dnsControl) sendServiceUpdates(s *api.Service) {
+	var z []string
+	for i := range dns.zones {
+		name := ServiceFQDN(s, dns.zones[i])
+		if _, ok := dns.watched[name]; ok {
+			z = append(z, name)
+		}
+	}
+	if len(z) > 0 {
+		*dns.watchChan <- z
+	}
+}
+
+func (dns *dnsControl) sendPodUpdates(p *api.Pod) {
+	var z []string
+	for i := range dns.zones {
+		name := PodFQDN(p, dns.zones[i])
+		if _, ok := dns.watched[name]; ok {
+			z = append(z, name)
+		}
+	}
+	if len(z) > 0 {
+		*dns.watchChan <- z
+	}
+}
+
+func (dns *dnsControl) sendEndpointsUpdates(ep *api.Endpoints) {
+	var z []string
+	for _, zone := range dns.zones {
+		names := append(EndpointFQDN(ep, zone, dns.endpointNameMode), ServiceFQDN(ep, zone))
+		for _, name := range names {
+			if _, ok := dns.watched[name]; ok {
+				z = append(z, name)
+			}
+		}
+	}
+	if len(z) > 0 {
+		*dns.watchChan <- z
```

Comment:
this needs to be queued, because .... the interface demands a `[]string`?
What's wrong with send z down where you do the `append`? Or is this prohibitive from a network traffic perspective?

Reply:
done